worlds/ladx/__init__.py

```diff
@@ -280,21 +291,49 @@ def create_items(self) -> None:
                     # Properly fill locations within dungeon
                     location.dungeon = r.dungeon_index
 
-        # For now, special case first item
-        FORCE_START_ITEM = True
-        if FORCE_START_ITEM:
-            self.force_start_item()
+        if self.options.tarins_gift != 'any_item':
+            self.force_start_item(itempool)
+
 
-    def force_start_item(self):    
+        self.multiworld.itempool += itempool
+
+    def force_start_item(self, itempool):
         start_loc = self.multiworld.get_location("Tarin's Gift (Mabe Village)", self.player)
         if not start_loc.item:
-            possible_start_items = [index for index, item in enumerate(self.multiworld.itempool)
-                if item.player == self.player 
-                    and item.item_data.ladxr_id in start_loc.ladxr_item.OPTIONS and not item.location]
-            if possible_start_items:
-                index = self.random.choice(possible_start_items)
-                start_item = self.multiworld.itempool.pop(index)
+            """
+            Find an item that forces progression or a bush breaker for the player, depending on settings.
+            """
+            if self.options.tarins_gift == 'local_progression':
+                base_collection_state = CollectionState(self.multiworld)
+                base_collection_state.update_reachable_regions(self.player)
+                reachable_count = len(base_collection_state.reachable_regions[self.player])
+
+            def is_possible_start_item(item):
+                return item.advancement and item.name not in self.options.non_local_items
+
+            def adheres_to_options(item):
+                if self.options.tarins_gift == 'local_progression':
+                    collection_state = base_collection_state.copy()
+                    collection_state.collect(item)
+                    return len(collection_state.reachable_regions[self.player]) > reachable_count
+                else: # bush breaker
+                    return item.name in self.item_name_groups['Bush Breakers']
+
+            possible_start_items = [item for item in itempool if is_possible_start_item(item)]
+            self.random.shuffle(possible_start_items)
+
+            start_item = None
+            for item in possible_start_items:
+                if adheres_to_options(item):
+                    start_item = item
+                    break

```

Comment:
Should be able to just:
```suggestion
            start_item = next((item for item in possible_start_items if adheres_to_options(item)), None)
```